### PR TITLE
Transfer-v2

### PR DESCRIPTION
**Pull request by piegamesde**

### CODE REVIEW

File: src/transfer.rs [hunk 1]
```diff
@@ -154,180 +158,677 @@ impl Default for AppVersion {
...
+        size: u64,
+        #[serde(skip)]
+        content: T,
+    },
+    Directory {
+        content: BTreeMap<String, Self>,
+    },
+    // Symlink {
+    //     target: String,
+    // },
```
Comment (wuan) [inline src/transfer.rs:530]: This still seems to be used from https://github.com/magic-wormhole/magic-wormhole.rs/blob/b51dcb21d508f00583f40684d3aac08292142c91/src/transfer/v1.rs#L316

**Comment by wuan** [comment]
And this call should be commented out/removed as well: https://github.com/magic-wormhole/magic-wormhole.rs/blob/b51dcb21d508f00583f40684d3aac08292142c91/cli/src/main.rs#L1101

**Comment by piegamesde** [comment]
Thanks. I'm a bit confused as to how I could have missed these, but also why the CI isn't running

**Review by wuan — APPROVED** [review]

File: src/core/test.rs [hunk 2]
```diff
@@ -35,227 +35,165 @@ fn default_relay_hints() -> Vec<transit::RelayHint> {
...
-            let wormhole = connector.await?;
-            eyre::Result::<_>::Ok(
-                transfer::send_file(
-                    wormhole,
-                    default_relay_hints(),
-                    &mut async_std::fs::File::open("tests/example-file-empty").await?,
-                    "example-file-empty",
-                    std::fs::metadata("tests/example-file-empty").unwrap().len(),
-                    magic_wormhole::transit::Abilities::ALL_ABILITIES,
+                req.accept(
```
[the remark this answers is not in the document]
Comment (piegamesde) [reply src/core/test.rs:184]: Given that v2 does not fully work yet it makes no sense to test it, therefore the tests are currently guaranteed to use v1 and v2 does not need to be handled for now.

**Review by vu3rdd — APPROVED** [review]
looks good to me.

File: src/core/test.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (wuan) [reply src/core/test.rs:184]: Then let's keep it like that for now.

**Pull request merged**